service/switchboardapp/src/main/java/com/switchboard/app/domain/Decoder.java

```diff
@@ -0,0 +1,27 @@
+package com.switchboard.app.domain;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.*;
+
+@Entity
+@AllArgsConstructor
+@NoArgsConstructor
+@Setter@Getter
+@JsonIgnoreProperties({"hibernateLazyIntializer","handler","device"})
+
+public class Decoder {
+
+    @Id
+    @Column(name = "serial_number")
+    private Long serialNumber;
```

Comment:
Add @NotNull unless that's already validated through the other annotations. If you do, add @Valid to the create new decoder endpoint.

Reply:
I think @id will fail in the db level if the value is null, adding @NotNull is also a good idea to fail at the method validation level

Reply:
fixed